core/domain/exp_services.py

```diff
@@ -1350,7 +1354,21 @@ def _save_exploration(
         exploration_model, exploration)
 
     change_list_dict = [change.to_dict() for change in change_list]
-    exploration_model.commit(committer_id, commit_message, change_list_dict)
+    commit_type = (
```

Comment:
As a general rule, please try not to ever copy and paste code from elsewhere in the codebase (especially anything with nontrivial "business logic"), and if you are doing that, make sure to call it out for the reviewer. This sort of thing is how we end up getting the skew that we are trying so hard to prevent.

Instead, the commit() method in VersionedModel should be refactored so that it calls another method which returns the models but doesn't put them, and this function should make use of that other method.